src/regressor.rs

```diff
@@ -398,14 +401,20 @@ impl Regressor {
         caches: &[BlockCache],
     ) -> f32 {
         pb.reset(); // empty the tape
-
+        let mut predictions_sum: f32 = 0.0;
         let further_blocks = &self.blocks_boxes[..];
-        block_helpers::forward_with_cache(further_blocks, fb, pb, caches);
-
-        assert_eq!(pb.observations.len(), 1);
-        let prediction_probability = pb.observations.pop().unwrap();
+        // First call with no dropout
+        block_helpers::forward_with_cache(further_blocks, fb, pb, caches, false);
```

Comment:
Since the FFM is the only thing that is impacted, instead of doing so many compute intensive operations, why don't we limit the scope to only doing something similar to a dropout only on FFM. Aka after calculating the contra_field pick a number between 1 & `number_of predictions` & multiple the `contra_field` with it & divide by  `number_of predictions`. Would give us a similar approximation without all the heavy duty calculations.